Modify asana stat calculators to work on top-level iterations and filter to team-specific ones

5outh opened the pull request:
By commit:

a01ef5f

This commit changes `close-iteration` to work only on the carryover
field in asana, but splits that carryover depending on the specific
_section_ it is in. In order to accomplish this, we now operate on the
top-level project (e.g. Iteration 152) with project id and supply the
name of the subproject to filter to for stat gathering.

For example,

```
stack exec -- close-iteration --project <project id> --subproject R152
```

Will provide stats for R152, provided that `<project id>` is for
Iteration 152.

For curriculum iterations , subproject need not be supplied, so it is
left as an optional argument.

e8253a1

This does the same thing as the previous commit, but for
`start-iteration`: We now operate on the top-level project and filter
down to subprojects to calculate stats.

5outh committed:
Split carryover field only in tooling

This commit changes `close-iteration` to work only on the carryover
field in asana, but splits that carryover depending on the specific
_section_ it is in. In order to accomplish this, we now operate on the
top-level project (e.g. Iteration 152) with project id and supply the
name of the subproject to filter to for stat gathering.

For example,

```
stack exec -- close-iteration --project <project id> --subproject R152
```

Will provide stats for R152, provided that <project id> matches up with
Iteration 152.

For curriculum iterations , subproject need not be supplied, so it is
left as an optional argument.

5outh committed:
Make start-iteration operate on top-level project

This does the same thing as the previous commit, but for
`start-iteration`: We now operate on the top-level project and filter
down to subprojects to calculate stats.

cdparks commented on library/Asana/Story.hs:
```diff
@@ -47,20 +47,31 @@ fromTask Task {..} = case tResourceSubtype of
       _ -> False
 
+  isCarryIn = flip any tMemberships $ \Membership {..} -> case mSection of
+    Just Named {..} | caseFoldEq nName "(101) Carryover" -> True
```
do we always have the `(101) ` here? can we do a

```haskell
"carryover" `Text.isInfixOf` Text.toLower nName
```

instead?

5outh committed:
Don't directly compare Carryover section name

5outh committed:
Infer commitment from carryover/cost

Commitment as state causes issues when it is updated too soon or too
late. Instead of saving this field and using it logically, this commit
updates it to be reconstructed from each task's cost and carry-in on the
fly.
